--- ./bootstrap
+++ ./bootstrap
@@ -54,6 +54,11 @@ cmake_system=`uname`
 cmake_source_dir=`cd "\`dirname \"$0\"\`";pwd`
 cmake_binary_dir=`pwd`
 
+if ${UNAME_OVERRIDE}; then
+  cmake_binary_dir="$(cygpath -m "${cmake_binary_dir}")"
+  cmake_source_dir="$(cygpath -m "${cmake_source_dir}")"
+fi
+
 # Load version information.
 cmake_version_major="`cmake_version_component MAJOR`"
 cmake_version_minor="`cmake_version_component MINOR`"
@@ -540,7 +545,7 @@ if ${cmake_system_darwin}; then
   "
 fi
 
-if ${cmake_system_mingw}; then
+if ${cmake_system_mingw} || ${UNAME_OVERRIDE}; then
   CMAKE_CXX_SOURCES="${CMAKE_CXX_SOURCES}\
     cmGlobalMSYSMakefileGenerator \
     cmGlobalMinGWMakefileGenerator \
@@ -575,7 +580,7 @@ LexerParser_C_SOURCES="\
   cmListFileLexer \
 "
 
-if ${cmake_system_mingw}; then
+if ${cmake_system_mingw} || ${UNAME_OVERRIDE}; then
   KWSYS_C_SOURCES="\
     EncodingC \
     ProcessWin32 \
@@ -633,7 +638,7 @@ JSONCPP_CXX_SOURCES="\
   src/lib_json/json_writer.cpp \
   "
 
-if ${cmake_system_mingw}; then
+if ${cmake_system_mingw} || ${UNAME_OVERRIDE}; then
   LIBUV_C_SOURCES="\
     src/fs-poll.c \
     src/idna.c
@@ -1655,6 +1660,9 @@ cmake_report cmConfigure.h${_tmp} "${cmake_compiler_settings_comment}"
 if ${cmake_system_mingw}; then
     CMAKE_BOOTSTRAP_SOURCE_DIR=`cd "${cmake_source_dir}"; pwd -W`
     CMAKE_BOOTSTRAP_BINARY_DIR=`cd "${cmake_binary_dir}"; pwd -W`
+elif ${cmake_system_cygwin}; then
+    CMAKE_BOOTSTRAP_SOURCE_DIR="$(cygpath -m "${cmake_source_dir}")"
+    CMAKE_BOOTSTRAP_BINARY_DIR="$(cygpath -m "${cmake_binary_dir}")"
 else
     CMAKE_BOOTSTRAP_SOURCE_DIR="${cmake_source_dir}"
     CMAKE_BOOTSTRAP_BINARY_DIR="${cmake_binary_dir}"
@@ -1682,7 +1690,7 @@ if ${cmake_system_darwin}; then
   cmake_report cmConfigure.h${_tmp} "#define CMake_USE_MACH_PARSER"
 fi
 
-if ${cmake_system_mingw}; then
+if ${cmake_system_mingw} || ${UNAME_OVERRIDE}; then
   cmake_report cmConfigure.h${_tmp} "#if defined(_WIN32) && !defined(NOMINMAX)"
   cmake_report cmConfigure.h${_tmp} "#  define NOMINMAX"
   cmake_report cmConfigure.h${_tmp} "#endif"
@@ -1763,7 +1771,7 @@ fi
 libs=""
 
 uv_c_flags=""
-if ${cmake_system_mingw}; then
+if ${cmake_system_mingw} || ${UNAME_OVERRIDE}; then
   uv_c_flags="${uv_c_flags} -DWIN32_LEAN_AND_MEAN -D_WIN32_WINNT=0x0600"
   libs="${libs} -lws2_32 -lpsapi -liphlpapi -lshell32 -luserenv -lole32 -loleaut32 -luuid"
 else
@@ -1799,7 +1807,7 @@ else
 fi
 if test "x${bootstrap_system_libuv}" = "x"; then
   uv_c_flags="${uv_c_flags} `cmake_escape_shell "-I${cmake_source_dir}/Utilities/cmlibuv/include"`"
-  if ${cmake_system_mingw}; then
+  if ${cmake_system_mingw} || ${UNAME_OVERRIDE}; then
     uv_c_flags="${uv_c_flags} `cmake_escape_shell "-I${cmake_source_dir}/Utilities/cmlibuv/src/win"`"
   else
     uv_c_flags="${uv_c_flags} `cmake_escape_shell "-I${cmake_source_dir}/Utilities/cmlibuv/src/unix"`"
@@ -1892,7 +1900,7 @@ write_source_rule() {
 }
 
 cmake_c_flags_String="-DKWSYS_STRING_C"
-if ${cmake_system_mingw}; then
+if ${cmake_system_mingw} || ${UNAME_OVERRIDE}; then
   cmake_c_flags_EncodingC="-DKWSYS_ENCODING_DEFAULT_CODEPAGE=CP_ACP"
   cmake_cxx_flags_EncodingCXX="${cmake_c_flags_EncodingC}"
   cmake_cxx_flags_cmProcessOutput="${cmake_c_flags_EncodingC}"
